contracts/src/Zappers/Modules/Exchanges/HybridCurveUniV3Exchange.sol

```diff
@@ -0,0 +1,163 @@
+        uint24 _feeUsdcWeth,
+        uint24 _feeWethColl,
+        ISwapRouter _uniV3Router
+    ) {
+        collToken = _collToken;
+        boldToken = _boldToken;
+        USDC = _usdc;
+        WETH = _weth;
+
+        // Curve
+        curvePool = _curvePool;
+        USDC_INDEX = _usdcIndex;
+        BOLD_TOKEN_INDEX = _boldIndex;
+
+        // Uniswap
+        feeUsdcWeth = _feeUsdcWeth;
+        feeWethColl = _feeWethColl;
+        uniV3Router = _uniV3Router;
+    }
+
+    // Bold -> USDC on Curve; then USDC -> WETH, and optionally WETH -> Coll, on UniV3
+    function swapFromBold(uint256 _boldAmount, uint256 _minCollAmount) external returns (uint256) {
+        InitialBalances memory initialBalances;
+        _setHybridExchangeInitialBalances(initialBalances);
+
+        // Curve
+        ICurveStableswapNGPool curvePoolCached = curvePool;
+        boldToken.transferFrom(msg.sender, address(this), _boldAmount);
+        boldToken.approve(address(curvePoolCached), _boldAmount);
+
+        uint256 curveUsdcAmount = curvePoolCached.exchange(int128(BOLD_TOKEN_INDEX), int128(USDC_INDEX), _boldAmount, 0);
+
+        // Uniswap
+        ISwapRouter uniV3RouterCached = uniV3Router;
+
+        USDC.approve(address(uniV3RouterCached), curveUsdcAmount);
+
+        // See: https://docs.uniswap.org/contracts/v3/guides/swaps/multihop-swaps
+        bytes memory path;
+        if (address(WETH) == address(collToken)) {
+            path = abi.encodePacked(USDC, feeWethColl, WETH);
+        } else {
+            path = abi.encodePacked(USDC, feeUsdcWeth, WETH, feeWethColl, collToken);
+        }
+
+        ISwapRouter.ExactInputParams memory params = ISwapRouter.ExactInputParams({
+            path: path,
+            recipient: msg.sender,
+            deadline: block.timestamp,
+            amountIn: curveUsdcAmount,
+            amountOutMinimum: _minCollAmount
+        });
+
+        // Executes the swap.
+        uint256 uniV3CollAmount = uniV3RouterCached.exactInput(params);
+
+        // return leftovers to user
+        _returnLeftovers(initialBalances);
+
+        return uniV3CollAmount;
+    }
+
+    // Optionally Coll -> WETH, and WETH -> USDC on UniV3; then USDC -> Bold on Curve
+    function swapToBold(uint256 _collAmount, uint256 _minBoldAmount) external returns (uint256) {
+        InitialBalances memory initialBalances;
+        _setHybridExchangeInitialBalances(initialBalances);
+
+        // Uniswap
+        ISwapRouter uniV3RouterCached = uniV3Router;
+
+        collToken.safeTransferFrom(msg.sender, address(this), _collAmount);
+        collToken.approve(address(uniV3RouterCached), _collAmount);
+
+        // See: https://docs.uniswap.org/contracts/v3/guides/swaps/multihop-swaps
+        bytes memory path;
+        if (address(WETH) == address(collToken)) {
+            path = abi.encodePacked(WETH, feeUsdcWeth, USDC);
+        } else {
+            path = abi.encodePacked(collToken, feeWethColl, WETH, feeUsdcWeth, USDC);
```

Comment:
Is this expected to be the best path (i.e. lowest slippage) specifically for RETH and WSTETH? Though maybe we're not trying to overly optimize that.

Reply:
Yes, that’s the assumption for now. Eventually we’ll create the dynamic version suggested by @danielattilasimon where the route can be passed as an argument.